libV2/helpers/collection/generateSkeletionTreeFromOpenAPI.js

```diff
@@ -46,94 +57,237 @@ let _ = require('lodash'),
             return;
           }
 
-          tree.setNode(`path:${pathSplit[0]}:${method}`, {
+          tree.setNode(`path:request:${pathSplit[0]}:${method}`, {
             type: 'request',
+            data: {},
             meta: {
-              path: path,
+              path: completePath,
               method: method,
               pathIdentifier: pathSplit[0]
-            },
-            data: {}
+            }
           });
 
-          tree.setEdge('root:collection', `path:${pathSplit[0]}:${method}`);
+          if (tree.hasNode(`path:folder:${pathSplit[0]}`)) {
+            tree.setEdge(`path:folder:${pathSplit[0]}`, `path:request:${pathSplit[0]}:${method}`);
+          }
+
+          else {
+            tree.setEdge('root:collection', `path:request:${pathSplit[0]}:${method}`);
+          }
         });
       }
 
       else {
-        _.forEach(pathSplit, function (p, index) {
+        _.forEach(pathSplit, function (path, index) {
           let previousPathIdentified = pathSplit.slice(0, index).join('/'),
             pathIdentifier = pathSplit.slice(0, index + 1).join('/');
 
-          /**
-           * Always first try to find the node if it already exists.
-           * if yes, bail out nothing is needed to be done.
-           */
-          if (tree.hasNode(`path:${pathIdentifier}`)) {
-            return;
+          if ((index + 1) === pathSplit.length) {
+            let methods = openapi.paths[completePath];
+
+            _.forEach(methods, function (data, method) {
+              if (!ALLOWED_HTTP_METHODS[method]) {
+                return;
+              }
+
+              /**
+               * include deprecated handling.
+               * If true, add in the postman collection. If false ignore the request.
+               */
+              if (!includeDeprecated && data.deprecated) {
+                return;
+              }
+
+              tree.setNode(`path:request:${pathIdentifier}:${method}`, {
+                type: 'request',
+                data: {},
+                meta: {
+                  path: completePath,
+                  method: method,
+                  pathIdentifier: pathIdentifier
+                }
+              });
+
+              /**
+               * If it is the last node,
+               * it might happen that this exists as a folder.
+               *
+               * If yes add a request inside that folder else
+               * add as a request on the previous path idendified which will be a folder.
+               */
+              if (tree.hasNode(`path:folder:${pathIdentifier}`)) {
+                tree.setEdge(`path:folder:${pathIdentifier}`, `path:request:${pathIdentifier}:${method}`);
+              }
+
+              else {
+                tree.setEdge(`path:folder:${previousPathIdentified}`, `path:request:${pathIdentifier}:${method}`);
```

Comment:
@sujayvenaik Why not create folder for last node if not already present? i.e This leaves case where suppose below are paths in spec.

```
team/{teamId}/user/{userId}
  get
  post
  put
team/{teamId}/user
  get
  post
team/{teamId}
  get
  post
```

For this, all requests from `team/{teamId}/user/{userId}` and `team/{teamId}/user` will remain in same folder. i.e. it'll have 5 requests.

Reply:
Tested this, working as expected now.